dbms/src/Flash/executeQuery.cpp

```diff
@@ -155,6 +155,7 @@ std::optional<QueryExecutorPtr> executeAsPipeline(Context & context, bool intern
     }
 
     prepareForExecute(context);
+    dag_context.enableResourceControl();
```

Comment:
Why change this

Reply:
We have to put it before PipelineExecutor is constructed.